src/rust/generators/osquery-subgraph-generator/src/generator.rs

```diff
@@ -0,0 +1,75 @@
+use async_trait::async_trait;
+
+use grapl_graph_descriptions::graph_description::*;
+use sqs_lambda::cache::{Cache, CacheResponse};
+use crate::metrics::OSQuerySubgraphGeneratorMetrics;
+use std::borrow::Cow;
+use log::*;
+use sqs_lambda::event_handler::{EventHandler, OutputEvent, Completion};
+use crate::parsers::PartiallyDeserializedOSQueryLog;
+use std::convert::TryFrom;
+
+
+#[derive(Clone)]
+pub(crate) struct OSQuerySubgraphGenerator<C>
+where
+    C: Cache + Clone + Send + Sync + 'static
+{
+    cache: C,
+    metrics: OSQuerySubgraphGeneratorMetrics,
+}
+
+impl<C> OSQuerySubgraphGenerator<C>
+where
+    C: Cache + Clone + Send + Sync + 'static
+{
+    pub fn new(cache: C, metrics: OSQuerySubgraphGeneratorMetrics) -> Self {
+        Self { cache, metrics }
+    }
+}
+
+#[async_trait]
+impl<C> EventHandler for OSQuerySubgraphGenerator<C>
+where
+    C: Cache + Clone + Send + Sync + 'static
+{
+    type InputEvent = Vec<PartiallyDeserializedOSQueryLog>;
+    type OutputEvent = Graph;
+    type Error = sqs_lambda::error::Error;
+
+    async fn handle_event(&mut self, input: Self::InputEvent) -> OutputEvent<Self::OutputEvent, Self::Error> {
+        info!("Processing {} incoming OSQuery log events.", input.len());
+
+        let (subgraphs, errors): (Vec<Option<Graph>>, Vec<Option<failure::Error>>) = input.into_iter()
+            .map(|log| {
+                match Graph::try_from(log) {
+                    Ok(graph) => (Some(graph), None),
+                    Err(e) => {
+                        warn!("Unable to convert partial OSQuery log into subgraph. {}", e);
```

Comment:
imo this should be `error!`

Reply:
Hmm the problem is that we actually expect a ton of these to fire. What happens is that we're parsing every possible osquery log into a generic `PartiallyDeserializedOSQueryLog` (the purpose is to read the log `name` field to determine what kind of log this is). 

We try to convert each of these into a `Graph`. Any log-types that aren't supported (e.g. any of the built-in OSQuery packs) will result in (an expected) failure. After that, some errors could occur if deserialization of supported logs fail or something but this will be less frequent.

Reply:
I'm not sure that finding a known and unsupported log type should result in a WARN, maybe not even Result in Error in code.